Scripts/Python/ki/xKIChat.py

```diff
@@ -582,6 +583,10 @@ def AddChatLine(self, player, message, cFlags, forceKI=True):
             else:
                 chatMessageFormatted = " {}".format(message)
 
+        chatMentions = []
+        if hasMention:
+            chatMentions = list(((i.start("mention"), i.end("mention"), i.group("mention")) for i in self._chatMentionRegex.finditer(chatMessageFormatted)))
```

Comment:
```suggestion
        if hasMention:
            chatMentions = ((i.start("mention"), i.end("mention"), i.group("mention")) for i in self._chatMentionRegex.finditer(chatMessageFormatted)))
        else:
            charMentions = ()
```

Generator comprehensions avoid allocating the entire list

Reply:
I needed it **not** to be a generator because it will be requested once and iterated twice (which I found out a couple days ago is not possible to do with an iterator/generator) -- for chatArea in (miniKI, microKI). I figured doing the .finditer once prior to that for-in loop might improve performance, thus the need for it to be iterable more than one time.

Reply:
That's right. This has to be iterated through for two chat areas. In that case, one small adjustment. Instead of creating a list from a generator comprehension `list((...))`, use a list comprehension instead `[...]`.

Reply:
Ok, I will do that. :) I will need to google python comprehensions because I don't understand it yet, though. Expect a push later today after work hours.

Reply:
Oh, I get it. So `chatMentions = [(i.start("mention"), i.end("mention"), i.group("mention")) for i in self._chatMentionRegex.finditer(chatMessageFormatted)]` will create a list from the iterable without going through the intermediate generator comprehension step that I had before, right?

Reply:
You've got it :+1:

Reply:
Committed adjustments for this and for your previous comments as well. I've finished kicking the tires on the regex (as well as added piece-by-piece comments for those who have to read that crap later), so I'm good with this all if you're good.